src/main/java/se/sundsvall/casemanagement/integration/ecos/RiskClassService.java

```diff
@@ -34,164 +34,171 @@
+		minutMiljoClient.addFacilityToCase(new AddFacilityToCase()
+			.withFacilityId(facilityId)
+			.withCaseId(caseId));
+
+	}
+
+	private SaveFoodFacility2024RiskClassData createSaveRiskClassObject(final String facilityId,
+		final String caseId, final EnvironmentalCaseDTO dto) {
+
+		return new SaveFoodFacility2024RiskClassData()
+			.withModel(new SaveRiskClass2024DataDto()
+				.withCaseId(caseId)
+				.withFacilityId(facilityId)
+				.withMainOrientationSlvCode(dto.getExtraParameters().get(MAIN_ORIENTATION_ID))
+				.withProductionSizeSlvCode(dto.getExtraParameters().get(PROD_SIZE_ID))
+				.withIsSeasonal(Optional.ofNullable(dto.getExtraParameters().get(IS_SEASONAL)).orElse("").equalsIgnoreCase("true"))
+				.withSeasonalNote(dto.getExtraParameters().get(SEASONAL_NOTE))
+				.withActivities(mapActivities(dto))
+				.withProductGroups(mapProductGroups(dto.getExtraParameters().get(PRODUCT_GROUPS)))
+				.withThirdPartyCertifications(mapThirdPartyCertifications(dto.getExtraParameters().get(THIRD_PARTY_CERTS))));
+	}
+
+	private ArrayOfSaveRiskClass2024ActivityDto mapActivities(final EnvironmentalCaseDTO dto) {
+		final StringBuilder activityString = new StringBuilder();
+
+		for (final var activityParam : ACTIVITIES) {
+
+			activityString.append(Optional.ofNullable(dto.getExtraParameters().get(activityParam)).orElse(" ")).append(
+				",");
+		}
+
+		if (activityString.isEmpty()) {
+			return null;
+		}
+		final var activities = splitString(activityString.toString());
+
+
+		return new ArrayOfSaveRiskClass2024ActivityDto()
+			.withSaveRiskClass2024ActivityDto(activities.stream()
+				.map(activitySlv -> new SaveRiskClass2024ActivityDto()
+					.withSlvCode(activitySlv))
+				.filter(activityDto -> !activityDto.getSlvCode().isEmpty())
+				.toList());
+	}
+
+	private ArrayOfSaveRiskClass2024ProductGroupDto mapProductGroups(final String productGroupIdString) {
+		if (productGroupIdString == null) {
+			return null;
+		}
+
+		final var productGroupIds = splitString(productGroupIdString);
+
+		if (productGroupIds.get(0).isEmpty()) {
+			return null;
+		}
+		return new ArrayOfSaveRiskClass2024ProductGroupDto()
+			.withSaveRiskClass2024ProductGroupDto(productGroupIds.stream()
+				.map(productGroupId -> new SaveRiskClass2024ProductGroupDto()
+					.withSlvCode(productGroupId))
+				.toList());
+	}
+
+	private ArrayOfSaveRiskClass2024CertificationDto mapThirdPartyCertifications(final String thirdPartyCertString) {
+		if (thirdPartyCertString == null) {
+			return new ArrayOfSaveRiskClass2024CertificationDto().withSaveRiskClass2024CertificationDto(new SaveRiskClass2024CertificationDto());
+		}
+		final var dtos = splitString(thirdPartyCertString);
+
+		return new ArrayOfSaveRiskClass2024CertificationDto()
+			.withSaveRiskClass2024CertificationDto(
+				dtos.stream()
+					.map(dto -> new SaveRiskClass2024CertificationDto()
+						.withThirdPartyCertificationText(dto))
+					.toList());
+	}
+
+	private List<String> splitString(final String string) {
```

Comment:
This is shorter and would give the same result, I think? :

`return Arrays.asList(string.split("(,+\\s?)"));`